src/java/org/apache/cassandra/io/sstable/format/SSTableFormat.java

```diff
@@ -36,6 +39,8 @@
     SSTableWriter.Factory getWriterFactory();
     SSTableReader.Factory getReaderFactory();
 
+    Set<Component> supportedComponents();
```

Comment:
As below, "supported" is not the right term here. Maybe "included"? (I believe we also have "required" for the mandatory ones?)

Reply:
I don't know, `supported` seemed to most relevant to me as it is intended to return all the components that can be loaded/used/supported by that format.

Reply:
OTOH, it can be also, something like `written`, or `produced` as this is also all the components that can be created when sstable is written.

Reply:
maybe will change this to just `all`